### PR TITLE
DA1469x SPI driver

### PR DESCRIPTION
**Pull request by blauret**
This adds the SPI driver for the Renesas SmartBond(tm) DA1469x MCU family. The driver only supports controller mode. All four SPI modes are supported. Note that the lowest supported speed is 2285714Hz. Requesting speeds higher than 16MHz, will result in a 16MHz SCLK.

The driver was tested on board with MCP4921 DAC and ADXL362 accelerometer

Signed-off-by: Ben Lauret <ben.lauret.wm@renesas.com>

### CODE REVIEW

File: boards/arm/da1469x_dk_pro/da1469x_dk_pro_defconfig [hunk 1]
```diff
@@ -13,3 +13,5 @@ CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 
 CONFIG_BUILD_OUTPUT_HEX=n
+
+CONFIG_SPI=y
```
Comment (gmarull) [inline boards/arm/da1469x_dk_pro/da1469x_dk_pro_defconfig:17]:
```suggestion
```

boards only need to enable basic peripherals, per porting guidelines

File: drivers/pinctrl/pinctrl_smartbond.c [hunk 2]
```diff
@@ -38,6 +38,9 @@ static void pinctrl_configure_pin(const pinctrl_soc_pin_t *pin)
 		reg_val |= 0x01 << GPIO_P0_00_MODE_REG_PUPD_Pos;
 	} else if (pin->bias_pull_down) {
 		reg_val |= 0x02 << GPIO_P0_00_MODE_REG_PUPD_Pos;
+	} else if (pin->output_enable) {
+		reg_val |= 0x03 << GPIO_P0_00_MODE_REG_PUPD_Pos;
+	} else if (pin->input_enable) {
 	}
```
Comment (gmarull) [inline drivers/pinctrl/pinctrl_smartbond.c:44]: no-op branch?

File: drivers/spi/spi_smartbond.c [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	}
+
+	return 0;
+}
+
+static int spi_smartbond_configure(const struct spi_smartbond_cfg *cfg,
+				   struct spi_smartbond_data *data,
+				   const struct spi_config *spi_cfg)
+{
+	int rc = 0;
```
Comment (gmarull) [inline drivers/spi/spi_smartbond.c:113]:
```suggestion
	int rc;
```

File: drivers/spi/spi_smartbond.c [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+{
+	const struct spi_smartbond_cfg *cfg = dev->config;
+	struct spi_smartbond_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	uint32_t bitmask;
+	int rc = 0;
+
+	spi_context_lock(&data->ctx, false, NULL, NULL, spi_cfg);
+	rc = spi_smartbond_configure(cfg, data, spi_cfg);
+	if (!(rc < 0)) {
```
Comment (gmarull) [inline drivers/spi/spi_smartbond.c:187]:
```suggestion
	if (rc == 0) {
```
suggestion

File: drivers/spi/spi_smartbond.c [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+static int spi_smartbond_transceive(const struct device *dev, const struct spi_config *spi_cfg,
+				    const struct spi_buf_set *tx_bufs,
+				    const struct spi_buf_set *rx_bufs)
+{
+	const struct spi_smartbond_cfg *cfg = dev->config;
+	struct spi_smartbond_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	uint32_t bitmask;
+	int rc = 0;
```
Comment (gmarull) [inline drivers/spi/spi_smartbond.c:183]:
```suggestion
	int rc;
```

File: dts/arm/renesas/smartbond/da1469x.dtsi [hunk 6]
```diff
@@ -84,6 +84,26 @@
 			interrupts = <5 0>;
 			status = "disabled";
 		};
+
+		spi: spi@50020300 {
+			compatible = "renesas,smartbond-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x50020300 0x100>;
+			periph-clock-config = <0x20>;
```
Comment (gmarull) [inline dts/arm/renesas/smartbond/da1469x.dtsi:93]: for future: this should use `clocks` property

**Review by gmarull — DISMISSED** [review]

File: dts/arm/renesas/smartbond/da1469x.dtsi (hunk 6 above)
Comment (blauret) [reply dts/arm/renesas/smartbond/da1469x.dtsi:93]: There's a coming PR with clock config. We'll keep in mind to update the peripherals

**Comment by carlescufi** [comment]
@tbursztyka could you please review?

**Comment by blauret** [comment]
@tbursztyka, @carlescufi,

Is there anything else needed to move forward?

File: boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml [hunk 7]
```diff
@@ -11,3 +11,4 @@ supported:
   - arduino_gpio
   - gpio
   - watchdog
+  - spi
```
Comment (teburd) [inline boards/arm/da1469x_dk_pro/da1469x_dk_pro.yaml:14]: Update the board docs as well please

**Review by teburd — DISMISSED** [review]
Seems like a simple enough polling driver, as noted please send a follow up PR updating the board docs, or push here and I'll re-approve

**Review by teburd — DISMISSED** [review]

**Comment by andrzej-kaczmarek** [comment]
SPI code looks good, but there are also changes to pinctrl driver so better split this into separate commits. Also there's some change in dac bindings which should be removed.

**Comment by carlescufi** [comment]
@blauret please rebase

File: dts/bindings/pinctrl/renesas,smartbond-pinctrl.yaml [hunk 8]
```diff
@@ -83,6 +83,8 @@ child-binding:
         property-allowlist:
           - bias-pull-down
           - bias-pull-up
+          - output-enable
+          - input-enable
```
Comment (andrzej-kaczmarek) [inline dts/bindings/pinctrl/renesas,smartbond-pinctrl.yaml:87]: pinctrl updates should better be moved to separate commit

File: dts/bindings/dac/microchip,mcp4921.yaml
Comment (andrzej-kaczmarek) [inline dts/bindings/dac/microchip,mcp4921.yaml:1]: this file does not seem to be related to SPI driver

**Review by andrzej-kaczmarek — CHANGES_REQUESTED** [review]

File: dts/bindings/dac/microchip,mcp4921.yaml
Comment (blauret) [reply dts/bindings/dac/microchip,mcp4921.yaml:1]: Actually we used this part to validate the SPI driver so we added the bindind

Comment (andrzej-kaczmarek) [reply dts/bindings/dac/microchip,mcp4921.yaml:1]: it's not related to the SPI driver itself and there's not even a matching driver for that part

Comment (blauret) [reply dts/bindings/dac/microchip,mcp4921.yaml:1]: Removed

File: dts/bindings/pinctrl/renesas,smartbond-pinctrl.yaml (hunk 8 above)
Comment (blauret) [reply dts/bindings/pinctrl/renesas,smartbond-pinctrl.yaml:87]: Done

**Review by tbursztyka — APPROVED** [review]

**Review by teburd — APPROVED** [review]
